tsdb/store.go

```diff
@@ -814,7 +814,9 @@ func (s *Store) DeleteShard(shardID uint64) error {
 	})
 
 	if err != nil {
-		s.Logger.Error("error walking shards during DeleteShard operation", zap.Error(err))
```

Comment:
I was initially concerned about removing this log message, even if it's not in 1.x. However, it appears that returning the here will cause the error to get logged eventually along all code paths.